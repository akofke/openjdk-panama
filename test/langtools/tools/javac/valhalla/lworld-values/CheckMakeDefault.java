/*
 * @test /nodynamiccopyright/
 * @summary Check various semantic constraints on value creation via __MakeDefault
 *
 * @compile/fail/ref=CheckMakeDefault.out -XDrawDiagnostics CheckMakeDefault.java
 */
__ByValue final class Point {

    __ByValue interface I {} // Error
    __ByValue abstract class A {} // Error
    static final class Sinner {
        static Sinner make() {
            return __MakeDefault Sinner(); // NO: Sinner is not a value class.
        }
    }

    __ByValue static final class SinnerValue {
        static SinnerValue make() {
            return __MakeDefault SinnerValue(); // OK.
        }
    }

    final int x;
    final int y;

    Point() {}
    Point (int x, int y) {}

    Point badFactory(int x, int y) {
        return __MakeDefault Point();
    }

    static Point make(int x, int y) {
       Point p = __MakeDefault Point(10, 20); // NO arguments to default value creation
       String s = __MakeDefault String(); // NO: String cannot be produced in this factory.
       __MakeDefault SinnerValue();
       p = __MakeDefault Point();
       p.x = x;
       p.y = y;
       return p;
    }
}
