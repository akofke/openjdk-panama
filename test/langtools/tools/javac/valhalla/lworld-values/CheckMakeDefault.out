CheckMakeDefault.java:9:15: compiler.err.illegal.combination.of.modifiers: interface, value
CheckMakeDefault.java:10:24: compiler.err.illegal.combination.of.modifiers: abstract, value
CheckMakeDefault.java:13:20: compiler.err.make.default.with.nonvalue
CheckMakeDefault.java:34:18: compiler.err.invalid.arguments.to.make.default
CheckMakeDefault.java:35:19: compiler.err.make.default.with.nonvalue
5 errors
